src/coreclr/src/vm/threadsuspend.cpp

```diff
@@ -6030,13 +5799,6 @@ void ThreadSuspend::SuspendEE(SUSPEND_REASON reason)
     if (reason == ThreadSuspend::SUSPEND_FOR_GC || reason == ThreadSuspend::SUSPEND_FOR_GC_PREP)
     {
         m_pThreadAttemptingSuspendForGC = pCurThread;
-
-        //
-        // also unblock any thread waiting around for this thread to suspend. This prevents us from completely
-        // starving other suspension clients, such as the debugger, which we otherwise would do because of
-        // the priority we just established.
```

Comment:
There are non-obvious interactions between thread suspend and debugger. It am not sure whether this is safe to delete.

Either way, it may be a good idea to run debugger tests on this.

Reply:
There is a single place where we wait on this event and it has a very short timeout. It timeouts often and waiter should be able to handle timeouts. Signaling the event, on the other hand, is an indication that there was some progress (a thread has suspended) and could be more confusing, since here there was no progress made.

Signaling probably made more sense in the past when suspension ping value was much longer.
 
I completely agree with the "debugger test" part. Should run some stress tests too.